libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -307,11 +308,20 @@ export function trigger({ element, event, type } = {}) {
 export const yieldToMain = () => new Promise((resolve) => { setTimeout(resolve, 0); });
 
 export async function fetchAndProcessPlainHtml({ url, shouldDecorateLinks = true } = {}) {
+  const { mep } = getConfig();
+
   const path = getFederatedUrl(url);
   const res = await fetch(path.replace(/(\.html$|$)/, '.plain.html'));
   const text = await res.text();
   const { body } = new DOMParser().parseFromString(text, 'text/html');
 
+  if (mep.globalnav.length) {
+    mep.globalnav.forEach(({ manifestId, commands }) => {
```

Comment:
Why does the globalnav need to call this method, rather than MEP calling it? 
Is it because MEP does not know when the nav is rendered?

Reply:
Yes because MEP does not know when the nav is rendered but also because the html MEP operates on never actually enters the DOM. The information is extracted from it and inserted into the relevant html templates we make for the nav. 
In an earlier iteration I had a message passing implementation that used Custom Events/Broadcast Channel but that ended up introducing even more coupling.